Subject: [PATCH] Align upstream handover test fixtures with the fresh flow
Base: ca8a66b2708ddab3022c2041e1beccf7c02bcaac

WHY: The fresh script-3 preflight now expects script 2 to have registered TAR
and assigned token CCIP administration to the DAO. Seed those inputs in the
existing upstream tests; behavioral regressions are maintained in the target.
This patch touches tests only and changes no deployment code.

diff --git a/chains/evm/contracts/lido-hvmv/test/script/deployment/SetPoolAndTransferOwnership.t.sol b/chains/evm/contracts/lido-hvmv/test/script/deployment/SetPoolAndTransferOwnership.t.sol
index e4ecbfc..a096dbe 100644
--- a/chains/evm/contracts/lido-hvmv/test/script/deployment/SetPoolAndTransferOwnership.t.sol
+++ b/chains/evm/contracts/lido-hvmv/test/script/deployment/SetPoolAndTransferOwnership.t.sol
@@ -4,6 +4,7 @@ pragma solidity ^0.8.24;
 import {PausableAdvancedPoolHooks} from "../../../PausableAdvancedPoolHooks.sol";
 import {PoolOperationManager} from "../../../PoolOperationManager.sol";
 import {SetPoolAndTransferOwnershipScript} from "../../../script/deployment/3_SetPoolAndTransferOwnership.s.sol";
+import {TokenAdminRegistry} from "@chainlink/contracts-ccip/tokenAdminRegistry/TokenAdminRegistry.sol";
 import {TokenPool} from "@chainlink/contracts-ccip/pools/TokenPool.sol";
 import {TokenAdminRegistry} from "@chainlink/contracts-ccip/tokenAdminRegistry/TokenAdminRegistry.sol";
 import {BurnMintERC20Transparent} from
@@ -16,6 +17,10 @@ import {Test} from "forge-std/Test.sol";
 
 // Injects the state that script 3 otherwise loads from the chain JSON.
 contract SetPoolAndTransferOwnershipHarness is SetPoolAndTransferOwnershipScript {
+  function setRegistry(address registry) external {
+    s_ccip.token_admin_registry = registry;
+  }
+
   function setTokenState(
     address deployer,
     address lidoDao,
@@ -95,6 +100,10 @@ contract SetPoolAndTransferOwnershipTest is Test {
     (s_token, s_proxyAdmin) = _deployToken(s_dao);
     s_script = new SetPoolAndTransferOwnershipHarness();
     s_script.setTokenState(address(this), s_dao, true, address(s_token), s_pool, s_proxyAdmin);
+    TokenAdminRegistry registry = new TokenAdminRegistry();
+    registry.proposeAdministrator(address(s_token), address(this));
+    registry.acceptAdminRole(address(s_token));
+    s_script.setRegistry(address(registry));
   }
 
   function _deployToken(
@@ -119,11 +128,13 @@ contract SetPoolAndTransferOwnershipTest is Test {
   }
 
   function test_checkDeployedTokenHandoverToDao_PassesWhilePending() public {
+    s_token.setCCIPAdmin(s_dao);
     s_token.beginDefaultAdminTransfer(s_dao);
     s_script.exposed_checkDeployedTokenHandoverToDao();
   }
 
   function test_checkDeployedTokenHandoverToDao_PassesAfterDaoAccepted() public {
+    s_token.setCCIPAdmin(s_dao);
     s_token.beginDefaultAdminTransfer(s_dao);
     vm.warp(block.timestamp + 1);
     vm.prank(s_dao);
@@ -143,6 +154,7 @@ contract SetPoolAndTransferOwnershipTest is Test {
 
   function test_RevertWhen_checkDeployedTokenHandoverToDao_ProxyAdminNotOwnedByDao() public {
     (BurnMintERC20Transparent token, address proxyAdmin) = _deployToken(address(this));
+    token.setCCIPAdmin(s_dao);
     token.beginDefaultAdminTransfer(s_dao);
     s_script.setTokenState(address(this), s_dao, true, address(token), s_pool, proxyAdmin);
 
